ddl/partition.go

```diff
@@ -280,3 +281,88 @@ func getPartitionIDs(table *model.TableInfo) []int64 {
 	}
 	return partitionIDs
 }
+
+// checkRangePartitioningKeysConstraints checks that the range partitioning key is included in the table constraint.
+func checkRangePartitioningKeysConstraints(ctx sessionctx.Context, s *ast.CreateTableStmt, tblInfo *model.TableInfo, constraints []*ast.Constraint) error {
+	// Returns directly if there is no constraint in the partition table.
+	if s.Partition.Tp != model.PartitionTypeRange || len(constraints) == 0 {
```

Comment:
Please move the "s.Partition.Tp != model.PartitionTypeRange" out of the function. It should be checked before entering the function.